src/code-checks/code-checks.service.ts

```diff
@@ -0,0 +1,324 @@
+            const volumetricUnitsOfMeasureCodeErrors = results[12].map(invalidCodeObj => getErrorMessage("volumetricUnitsOfMeasureCode", invalidCodeObj.column1))
+            const sourceOfDataVolumetricCodeErrors = results[13].map(invalidCodeObj => getErrorMessage("sourceOfDataVolumetricCode", invalidCodeObj.column1))
+            const sourceOfDataMassCodeErrors = results[14].map(invalidCodeObj => getErrorMessage("sourceOfDataMassCode", invalidCodeObj.column1))
+            const sampleTypeCodeErrors = results[15].map(invalidCodeObj => getErrorMessage("sampleTypeCode", invalidCodeObj.column1))
+            const parameterUnitsOfMeasureCodeErrors = results[16].map(invalidCodeObj => getErrorMessage("parameterUnitsOfMeasureCode", invalidCodeObj.column1))
+            const beginEndHourFlagErrors = results[17].map(invalidCodeObj => getErrorMessage("beginEndHourFlag", invalidCodeObj.column1))
+            const fuelFlowPeriodCodeErrors = results[18].map(invalidCodeObj => getErrorMessage("fuelFlowPeriodCode", invalidCodeObj.column1))
+            const longTermFuelFlowUnitsOfMeasureCodeErrors = results[19].map(invalidCodeObj => getErrorMessage("longTermFuelFlowUnitsOfMeasureCode", invalidCodeObj.column1))
+            const gcvUnitsOfMeasureCodeErrors = results[20].map(invalidCodeObj => getErrorMessage("gcvUnitsOfMeasureCode", invalidCodeObj.column1))
+            const apsCodeErrors = results[21].map(invalidCodeObj => getErrorMessage("apsCode", invalidCodeObj.column1))
+            const samplingRatioCheckResultCodeErrors = results[22].map(invalidCodeObj => getErrorMessage("samplingRatioCheckResultCode", invalidCodeObj.column1))
+            const postLeakCheckResultCodeErrors = results[23].map(invalidCodeObj => getErrorMessage("postLeakCheckResultCode", invalidCodeObj.column1))
+            const trainQAStatusCodeErrors = results[24].map(invalidCodeObj => getErrorMessage("trainQAStatusCode", invalidCodeObj.column1))
+            const co2EmissionStandardCodeErrors = results[25].map(invalidCodeObj => getErrorMessage("co2EmissionStandardCode", invalidCodeObj.column1))
+            const modusUnitsOfMeasureCodeErrors = results[26].map(invalidCodeObj => getErrorMessage("modusUnitsOfMeasureCode", invalidCodeObj.column1))
+            const electricalLoadCodeErrors = results[27].map(invalidCodeObj => getErrorMessage("electricalLoadCode", invalidCodeObj.column1))
+            const annualEnergySoldTypeCodeErrors = results[28].map(invalidCodeObj => getErrorMessage("annualEnergySoldTypeCode", invalidCodeObj.column1))
+            const co2EmissionRateUnitsOfMeasureCodeErrors = results[29].map(invalidCodeObj => getErrorMessage("co2EmissionRateUnitsOfMeasureCode", invalidCodeObj.column1))
+
+            errorList.push(...parameterCodeErrors, ...fuelCodeErrors, ...testTypeCodeErrors, ...testResultCodeErrors, ...spanScaleCodeErrors, ...gasLevelCodeErrors,
```

Comment:
you can then use a loop on the promises array to spread the error messages which makes this a little more readable. Something like below...

promises.forEach(promise in promises) {
    errorList.push(...promises);
}